Docs / How to document

srosset81 opened the pull request:
Je remets ici le travail de Guillaume, qui avait été accidentellement mergé sur master.

Les commentaires de Gabriel sont toujours visibles ici: https://github.com/assemblee-virtuelle/semapps/pull/204

@GuillaumeAV et @ReX-AV-Gab je vous laisse vous coordonner pour finaliser cette PR, notamment en ajoutant un lien dans la sidebar. Une fois que ce sera fait, vous pourrez me mettre comme reviewer.

srosset81 committed:
Revert Guillaume's work

GuillaumeAV commented on website/docs/participate/documentation.md:
```diff
@@ -0,0 +1,44 @@
+  - On y trouve les pages apparaissant dans la sidebar
+  
+# Comment contribuer à la documentation
+
```
## Schéma général
- Tant qu'il n'y a pas eu de décision / consentement du cercle documentation : **On renseigne une issue**
- Si le contenu de ce qu'on veut publier est à l'état d'ébauche ou au stade de brouillon : **On renseigne une issue**
- Lorsque notre proposition de contribution a obtenu l'aval du groupe documentation, et qu'elle nous semble aboutie, on pousse une pull request.

GuillaumeAV commented on website/docs/participate/documentation.md:
````diff
@@ -0,0 +1,44 @@
+- Créez un fichier markdown à l'endroit souhaité.Celui-ci doit nécessairement commencer par :
+  ```
+  ---
+  title: [le titre de votre page]
````
title: [le titre-de-votre-page]

Commit:
Page d'accueil

GuillaumeAV committed:
Update documentation.md

srosset81 committed:
Update sidebars.js

srosset81 committed:
Revert homepage

GuillaumeAV committed:
Schéma général + Modifications mineures

Update du schéma général, avec recommandation de créer un pull request, une fois l'aval du GT acquis .. 
Orthographe, tournures de phrases .

GuillaumeAV committed:
Ajout d'une précision à propos des commentaires

GuillaumeAV committed:
Update documentation.md

GuillaumeAV committed:
MàJ du doc pour intégrations des suggestions de Gabriel

Voir les commentaires ci-dessus ..

GuillaumeAV reviewed (approved):
Ok pour moi !

GuillaumeAV commented:
Yop ! 
@srosset81 je viens d'approuver la PR et Gab qui est au tel avec moi, l'approuve aussi, tu merges du coup ? La bise !

srosset81 commented on website/docs/participate/documentation.md:
```diff
@@ -0,0 +1,55 @@
+# Introduction 
+- Nous utilisons https://docusaurus.io/ pour documenter le projet SemApps. Docusaurus permet de générer une documentation à partir de Github : 
```
https://v2.docusaurus.io ... et je mettrai des jolis liens

srosset81 commented on website/docs/participate/documentation.md:
```diff
@@ -0,0 +1,55 @@
+    - On bénéficie donc de l'environnement collaboratif github (édition markdown, versionning, commits, pull requests, reviews, merge etc.)
+    - On peut prévisualiser les changements
+    - On peut documenter le code et afficher la documentation sur le site vitrine.
+- Aujourd'hui le docusaurus de SemApps est hébergé sur https://website-3rr9j6b6k.now.sh/ ; 
```
La version temporaire est plutôt http://doc.semapps.org/

srosset81 commented on website/docs/participate/documentation.md:
```diff
@@ -0,0 +1,55 @@
+- https://github.com/assemblee-virtuelle/semapps/blob/master/website/src/pages/index.js
+  - On y trouve le contenu du corps de la page d'accueil et également le texte du bouton (Commencer)
+- https://github.com/assemblee-virtuelle/semapps/blob/master/website/sidebars.js
+  - On y trouve les pages apparaissant dans la sidebar
```
dans **les** sidebars, car on peut définir plusieurs sidebars.

srosset81 commented on website/docs/participate/documentation.md:
```diff
@@ -0,0 +1,55 @@
+- Si le contenu de ce qu'on veut publiera obtenu l'aval du groupe documentation : 
+    - **On crée une pull request sur une branche dédiée.**  
+    - Tant qu'il est à l'état d'ébauche ou au stade de brouillon, **on ne demande pas de review**
+- Lorsque notre proposition de contribution nous semble aboutie, **on sollicite ue review**.
```
Une review. Et au dessus: **a** obtenu.
(Merci de relire attentivement une PR avant de la soumettre à review, cela facilite le travail)

srosset81 commented on website/docs/participate/documentation.md:
```diff
@@ -0,0 +1,55 @@
+  - Renseignez un __Label__ et identifiez des __Reviewers__ qui seront invités à commenter / approuver / proposer des modifications à votre pull request. 
+  - A l'issue de ce processus collectif, viendra le moment de __merger__ la pull request sur la branche __Master__
+  
+NB : Lorsque que le fichier a été créé par un autre et en fonction du contexte, vous pouvez soit éditer la page, soit cliquer sur les signes "+" à gauche du texte pour ajouter des commentaires ... 
```
Ajouter un commentaire en cliquant sur le signe "+" à gauche de la ligne est pertinent surtout pour des reviews, je ne sais pas si je le mettrai ici.

srosset81 commented on website/docs/participate/documentation.md:
````diff
@@ -0,0 +1,55 @@
+- Créez un fichier markdown à l'endroit souhaité, en n'oubliant pas de renseigner l'extention : ".md". Celui-ci doit nécessairement commencer par :
+  ```
+  ---
+  title: [le-titre-de-votre-page]
````
Le titre de la page peut avoir des espaces. C'est le nom du fichier `.md` qui ne doit pas avoir d'espaces, et être plutôt en minuscules.

srosset81 reviewed (commented):
Comme j'ai créé cette PR, je ne peux pas la reviewer mais voici mes commentaires.
Remarque générale: n'avez-vous pas décidé entre temps d'écrire toute la doc externe en anglais ?

GuillaumeAV committed:
Mise à jour du doc suite aux commentaires de Seb 

Avant traduction en anglais ;-)

GuillaumeAV committed:
English translation !

srosset81 committed:
Merge branch 'master' into docs/how-to-document

# Conflicts:
#	website/sidebars.js
